src/main/java/com/github/technus/tectech/thing/metaTileEntity/multi/GT_MetaTileEntity_EM_EyeOfHarmony.java

```diff
@@ -0,0 +1,2013 @@
+            if (processRecipe(currentRecipe)) {
+                return true;
+            }
+
+            currentRecipe = null;
+        }
+        return false;
+    }
+
+    private long getHydrogenStored() {
+        return validFluidMap.get(Materials.Hydrogen.getGas(1));
+    }
+
+    private long getHeliumStored() {
+        return validFluidMap.get(Materials.Helium.getGas(1));
+    }
+
+    public boolean processRecipe(EyeOfHarmonyRecipe recipeObject) {
+
+        // Debug mode, overwrites the required fluids to initiate the recipe to 100L of each.
+        if (EOHDebugMode) {
+            if ((getHydrogenStored() < 100) || (getHeliumStored() < 100)) {
+                return false;
+            }
+        } else {
+            if ((getHydrogenStored() < currentRecipe.getHydrogenRequirement())
+                    || (getHeliumStored() < currentRecipe.getHeliumRequirement())) {
+                return false;
+            }
+        }
+
+        // Check tier of spacetime compression blocks is high enough.
+        if (spacetimeCompressionFieldMetadata < recipeObject.getSpacetimeCasingTierRequired()) {
+            return false;
+        }
+
+        // Remove EU from the users network.
+        if (!addEUToGlobalEnergyMap(userUUID, -recipeObject.getEUStartCost())) {
+            return false;
+        }
+
+        mMaxProgresstime = recipeProcessTimeCalculator(
+                recipeObject.getRecipeTimeInTicks(), recipeObject.getSpacetimeCasingTierRequired());
+
+        calculateHydrogenHeliumInputExcessValues(
+                recipeObject.getHydrogenRequirement(), recipeObject.getHeliumRequirement());
+
+        if (EOHDebugMode) {
+            hydrogenOverflowProbabilityAdjustment = 0;
+            heliumOverflowProbabilityAdjustment = 0;
+        }
+
+        successChance = recipeChanceCalculator();
+
+        // Determine EU recipe output.
+        euOutput = recipeObject.getEUOutput();
+
+        // Reduce internal storage by hydrogen and helium quantity required for recipe.
+        validFluidMap.put(Materials.Hydrogen.getGas(1), 0L);
+        validFluidMap.put(Materials.Helium.getGas(1), 0L);
+
+        double yield = recipeYieldCalculator();
+        if (EOHDebugMode) {
+            successChance = 1; // Debug recipes, sets them to 100% output chance.
+        }
+
+        // Return copies of the output objects.
+        mOutputFluids = recipeObject.getOutputFluids();
+        outputItems = recipeObject.getOutputItems();
+
+        if (yield != 1.0) {
+            // Iterate over item output list and apply yield values.
+            for (ItemStackLong itemStackLong : outputItems) {
+                itemStackLong.stackSize *= yield;
+            }
+
+            // Iterate over fluid output list and apply yield values.
+            for (FluidStack fluidStack : mOutputFluids) {
+                fluidStack.amount *= yield;
+            }
+        }
+
+        updateSlots();
+
+        createRenderBlock();
```

Comment:
This is no real TT machine if it doesn't explode when this block is obstructed :trollface: